x-pack/plugins/apm/public/hooks/useFetcher.tsx

```diff
@@ -46,8 +44,9 @@ export function useFetcher<Opts, Response>(
       }
     });
 
-    fn(fnOptions)
-      .then(data => {
+    async function doFetch() {
+      const data = await fn();
```

Comment:
Doesn't this line need to be inside the `try {}`?

Reply:
I think with how the try/catch works you could do the `didFinish` just once at the end if you want:
```js
try {
  const data = await fn();
  // do success stuff
} catch (e) {
  // do error stuff
}
didFinish = true;
```
Doesn't hurt to do it twice if you think that's clearer.

Reply:
Ahh, yes. Must have slipped when I converted from promises.